pkg/ssh/utils.go

```diff
@@ -21,6 +21,7 @@ func Validate(user *url.Userinfo, path string, port int, identity string) (*conf
 	if strings.Contains(path, "/run") {
 		sock = strings.Split(path, "/run")[1]
 	}
+	// url.Parse NEEDS ssh://, if this ever fails or returns some nonsense, that is why.
```

Comment:
Looking at the “sometimes we are not going to have a path” comment below?
- What’s an example? I guessed `ssh://foo` and `ssh://user@foo`, but that didn’t match the code. If there is a a corner case (`ssh:foo`??), it should be documented clearly — perhaps in a comment, but ideally actually documented _also_ by a unit test that ensures the behavior is not accidentally broken.
- How exactly does “not going to have a path” relate to the presence, or not, of a user designation? I can‘t see the relationship.